platform-sdk/docs/proposals/healthMonitor/healthMonitorProposal.md

```diff
@@ -0,0 +1,127 @@
+# Health Monitor Proposal
+
+---
+
+## Summary
+
+A replacement for the current hard backpressure mechanism using a health monitor.
+
+| Metadata           | Entities                            | 
+|--------------------|-------------------------------------|
+| Designers          | Cody Littley                        |
+| Functional Impacts | Consensus node performance          |
+| Related Proposals  | n/a                                 |
+| HIPS               | n/a                                 |
+
+---
+
+## Purpose and Context
+
+The platform's current backpressure mechanism has very bad performance characteristics.
+We need to replace the current backpressure implementation with something more performant.
+
+### Requirements
+
+- The backpressure mechanism must prevent an OOM in the event of any arbitrary component
+  becoming arbitrarily slow.
+- The backpressure mechanism should not slow down critical processes when enganged.
+  Specifically, transaction handling should not be impeded when backpressure is engaged.
```

Comment:
I agree. The new backpressure mechanism doesn't treat transaction handling differently. But since transaction handling is currently the system bottleneck, I think it's important to call it out specifically in the acceptance criteria for this change.